src/engine/Eval_generic.ml

```diff
@@ -246,18 +284,30 @@ and eval_op op values code =
   | G.Gt, [ Float i1; Float i2 ] -> Bool (i1 > i2)
   | G.Gt, [ Int i1; Float i2 ] -> Bool (float_of_int i1 > i2)
   | G.Gt, [ Float i1; Int i2 ] -> Bool (i1 > float_of_int i2)
+  | G.Gt, [ Date (y1, m1, d1); Date (y2, m2, d2) ] ->
+      Bool (date_compare (y1, m1, d1) (y2, m2, d2) =*= 1)
   | G.GtE, [ Int i1; Int i2 ] -> Bool (i1 >= i2)
   | G.GtE, [ Float i1; Float i2 ] -> Bool (i1 >= i2)
   | G.GtE, [ Int i1; Float i2 ] -> Bool (float_of_int i1 >= i2)
   | G.GtE, [ Float i1; Int i2 ] -> Bool (i1 >= float_of_int i2)
+  | G.GtE, [ Date (y1, m1, d1); Date (y2, m2, d2) ] ->
```

Comment:
if the date was a float, you would not need all those extra cases. The case for Float would just work.

Reply:
I think the problem with this is that the user requested being able to do something like date($X) where $X is a string of (for example) format "2023-04-15" not just a float. If treating dates as just floats, then would still need to do conversions from these strings into floats. Could represent with just floats, but it will still require a good amount of conversion to do what the user wanted either way. I'm also worried that if we just use a float, and then convert these strings to a float representation then we lose timezones.

Reply:
Or not lose timezones, but need to adjust more to timezones, happy to change to floats though if that is the consensus